mdfreader/mdf4reader.py

```diff
@@ -1363,14 +1394,41 @@ def returnField(obj, field):
         if self._noDataLoading and channel_list is not None:
             data_groups = [self[channel][idField][0][0] for channel in channel_list]
 
+        if source_list is not None:
+            source_dataFrame = pd.DataFrame(source_list, columns=['source_list'])
+            # there is just a source occurrence
+            keep_first_source = list(source_dataFrame.drop_duplicates(keep="first")['source_list'])
+        own_source = str()
+
         for dataGroup in data_groups:
             channel_set = channel_set_file
             if not info['DG'][dataGroup]['dg_data'] == 0 and \
                     (channel_set is None or
                      len(channel_set & info['ChannelNamesByDG'][dataGroup]) > 0):  # there is data block and channel in
+                if source_list is not None:
+                    find_own_source = False
+                    skip_this_group = False
+                    # if there is no signal in the channel group
+                    if len(channel_set & info['ChannelNamesByDG'][dataGroup]) < 2 and 't' in list(
```

Comment:
It could be t but also time, Time or another, not so robust

Reply:
I've made an adjustment concerning the master channel(time in my acquisitions), aiming for a unique name for that channel. Specifically, I replaced it with "self['masters'][self['CG'][dg-1][cg]['pointer']]['name']". I would value your insights and thoughts on this modification. If you have any suggestions or considerations regarding this change, please feel free to share.